packages/java/parser-jvm-test-utils/src/main/java/com/vaadin/hilla/parser/testutils/JsonAssertions.java

```diff
@@ -11,7 +13,35 @@ public final class JsonAssertions {
 
     public static void assertEquals(Object expected, Object actual)
             throws JsonProcessingException {
-        Assertions.assertEquals(printer.pretty().writeAsString(expected),
-                printer.pretty().writeAsString(actual));
+        try {
+            String expectedJson = printer.writeAsString(expected);
+            String actualJson = printer.writeAsString(actual);
```

Comment:
I would move this outside the try catch used for the assertion. It should simplify debugging if something fails with the `printer` and it allows enhancing the JSON assertion message by directly providing the compared JSON strings instead of having to serialize the objects again,